frontend/__tests__/UserSignUp.test.jsx

```
@@ -0,0 +1,260 @@
+    fireEvent.click(screen.getByText('Sign Up'));
+
+    await waitFor(() => {
+      expect(screen.getByText('Something went wrong. Please try again.')).toBeInTheDocument();
+      expect(console.error).toHaveBeenCalled();
+      expect(mockNavigate).not.toHaveBeenCalled();
+    });
+  });
+
+  test('redirects to Google OAuth when clicking Google signup button', () => {
+    const locationAssignMock = vi.fn();
+    delete window.location;
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)